plugins/lightspeed-backend/src/service/router.ts

```diff
@@ -0,0 +1,74 @@
+import { errorHandler } from '@backstage/backend-common';
+import { LoggerService } from '@backstage/backend-plugin-api';
+import { Config } from '@backstage/config';
+
+import express from 'express';
+import Router from 'express-promise-router';
+import { APIError, OpenAI } from 'openai';
+
+import { CompletionsRequestBody } from './types';
+import { validateCompletionsRequest } from './validation';
+
+export interface RouterOptions {
+  logger: LoggerService;
+  config: Config;
+}
+
+export async function createRouter(
+  options: RouterOptions,
+): Promise<express.Router> {
+  const { logger, config } = options;
+
+  const router = Router();
+  router.use(express.json());
+
+  router.get('/health', (_, response) => {
+    logger.info('PONG!');
+    response.json({ status: 'ok' });
+  });
+
+  router.post(
+    '/chat/completions',
+    validateCompletionsRequest,
+    async (request, response) => {
+      const { model, messages }: CompletionsRequestBody = request.body;
+      const baseURL = config.getString('lightspeed.baseURL');
+      const apiKey = config.getString('lightspeed.apiKey');
+
+      const openai = new OpenAI({
+        baseURL: baseURL,
+        apiKey: apiKey,
+      });
+
+      try {
+        const stream = await openai.chat.completions.create({
```

Comment:
Wouldn't it be better if we just do proxying of requests from UI to the API server? That way we can utilize any OpenAI's complaint API from the UI directly. We can build other APIs as needed into the backend like the chat history or something like that but I am not sure if we should wrap the completion API like this.

Reply:
Yes, that could be another approach. And then sending messages to save in separate API calls. We should discuss this and then redefine [RHIDP-2949](https://issues.redhat.com/browse/RHIDP-2949) if needed.